File: 206 Reverse Linked List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        if(!head || !head->next)
            return head;
        auto re = head;
        while(head){
            data.push_back(head->val);
            head = head->next;
        }
        auto w = re;
        for(int i = data.size()-1;i >= 0;--i){
            w->val = data[i];
            w = w->next;
        }
        return re;
    }
private:
    vector<int> data;//stack意外的比較慢
};

File: 206 Reverse Linked List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    let arr = [];
    arr.push(null);
    let cur = head;
    while(cur !== null){
        arr.push(cur);
        cur = cur.next;
    }
    for(let i = arr.length - 1;i > 0;i--){
        arr[i].next = arr[i-1];
    }
    return arr[arr.length - 1];
};
